pkg/vm/opcodebench_test.go

```diff
@@ -74,714 +75,1578 @@ func opParamSlotsPushVM(op opcode.Opcode, param []byte, sslot int, slotloc int,
-	var m = stackitem.NewMap()
-	for i := range 1024 {
-		m.Add(stackitem.Make(i), stackitem.Make(i))
+func BenchmarkConvert(b *testing.B) {
+	type testCase struct {
+		name   string
+		params []byte
+		items  []any
+	}
+	var (
+		testCases        []testCase
+		stepArrayLen     = MaxStackSize / countPoints
+		stepByteArrayLen = stackitem.MaxSize / countPoints
+		currArrayLen     = stepArrayLen - 1
+		currByteArrayLen = stepByteArrayLen
+	)
+	for range countPoints {
+		testCases = append(testCases, testCase{
+			name:   fmt.Sprintf("%s %s %s %d", opcode.CONVERT, stackitem.ArrayT, stackitem.StructT, currArrayLen),
+			params: []byte{byte(stackitem.StructT)},
+			items:  []any{make([]any, currArrayLen)},
+		})
+		currArrayLen += stepArrayLen
+	}
+	for range countPoints {
+		testCases = append(testCases, testCase{
+			name:   fmt.Sprintf("%s %s %s %d", opcode.CONVERT, stackitem.ByteArrayT, stackitem.BufferT, currByteArrayLen),
+			params: []byte{byte(stackitem.BufferT)},
+			items:  []any{stackitem.NewByteArray(make([]byte, currByteArrayLen))},
+		})
+		currByteArrayLen += stepByteArrayLen
+	}
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			benchOpcode(b, opParamPushVM(opcode.CONVERT, tc.params, tc.items...))
+		})
 	}
-	return m
 }
 
-func maxBytes() []byte {
-	return zeroSlice(1024 * 1024)
+func BenchmarkInitSlot(b *testing.B) {
+	type testCase struct {
+		name   string
+		op     opcode.Opcode
+		params []byte
+		items  []any
+	}
+	var (
+		testCases []testCase
+		stepSlots = byte(math.MaxUint8 / countPoints)
+		currSlots = stepSlots
+	)
+	// TODO: do not ignore the len and depth of the argument
+	for range countPoints {
+		for _, params := range []struct{ localSlots, argSlots byte }{{currSlots, 0}, {0, currSlots}} {
+			testCases = append(testCases, testCase{
+				name:   fmt.Sprintf("%s %d %d", opcode.INITSLOT, params.localSlots, params.argSlots),
+				op:     opcode.INITSLOT,
+				params: []byte{params.localSlots, params.argSlots},
+				items:  make([]any, params.argSlots),
+			})
+		}
+		testCases = append(testCases, testCase{
+			name:   fmt.Sprintf("%s %d", opcode.INITSSLOT, currSlots),
+			op:     opcode.INITSSLOT,
+			params: []byte{currSlots},
+		})
+		currSlots += stepSlots
+	}
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			benchOpcode(b, opParamPushVM(tc.op, tc.params, tc.items...))
+		})
+	}
 }
 
-func maxBuf() *stackitem.Buffer {
-	return stackitem.NewBuffer(maxBytes())
+func BenchmarkLD(b *testing.B) {
+	type testCase struct {
+		name string
+		f    func() *VM
+	}
+	var (
+		testCases []testCase
+		stepDeep  = MaxStackSize / countPoints
+		currDeep  = stepDeep - 1
+	)
+	for range countPoints {
+		item := stackitem.NewArray(getArrayWithDeep(currDeep))
+		testCases = append(testCases, testCase{
+			name: fmt.Sprintf("%s %d", opcode.LDSFLD0, currDeep),
+			f: func() *VM {
+				v := opVM(opcode.LDSFLD0)()
+				slot := []stackitem.Item{item}
+				v.Context().sc.static = slot
```

Comment:
It's a common case: don't introduce new variable if it's not needed, it makes the code less readable.
```suggestion
				v.Context().sc.static = []stackitem.Item{item}
```